src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -159,6 +164,11 @@ void Run (DirectoryAssemblyResolver res)
 				}
 
 				res.Load (assembly.ItemSpec);
+#if ENABLE_MARSHAL_METHODS
+				if (!Debug) {
```

Comment:
Why not skip these `if (!Debug) StoreMarshalAssemblyPath()` checks and just call `StoreMarshalAssemblyPath())`?  `StoreMarshalAssemblyPath()` checks `Debug` anyway, so this is just adding extra levels of complexity & indentation, for no observable benefit.

Reply:
In the first case we save one string allocation, in the second instance yes, it's not necessary